### PR TITLE
Linter setup

### PR DESCRIPTION
**Pull request by illabo**
Xcode Run Script build phase added to utilize systemwide installation of SwiftLint if it is installed.
Some minimal SwiftLint settings added to start discussion.
For discussion check #13

### CODE REVIEW

**Review by illabo — COMMENTED** [review]
During the Run Script build phase in case SwiftLint were not installed in system a local copy for project would be downloaded (once only) and used for linting. 
```bash
# Lint with systemwide installation if present
if which swiftlint >/dev/null; then
  swiftlint
  exit 0
fi
# Lint with local swiftlint copy if present
if [ -f ${PROJECT_DIR}/swiftlint ]; then
  ${PROJECT_DIR}/swiftlint
else
  # Download a copy of swiftlint to project root and lint afterwards
  curl -o- -Ls https://github.com/realm/SwiftLint/releases/latest/download/portable_swiftlint.zip | tar xzf - swiftlint \
  && chmod +x swiftlint || echo "warning: Failed to download SwiftLint"; exit 0
  ${PROJECT_DIR}/swiftlint 
fi
```

File: FlipperZero/FlipperZero.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -376,6 +378,45 @@
...
+			inputPaths = (
+			);
+			name = "Check code with SwiftLint";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "# Lint with systemwide installation if present\nif which swiftlint >/dev/null; then\n  swiftlint\n  exit 0\nfi\n# Lint with local swiftlint copy if present\nif [ -f ${PROJECT_DIR}/swiftlint ]; then\n  ${PROJECT_DIR}/swiftlint\nelse\n  # Download a copy of swiftlint to project root and lint afterwards\n  curl -o- -Ls https://github.com/realm/SwiftLint/releases/latest/download/portable_swiftlint.zip | tar xzf - swiftlint \\\n  && chmod +x swiftlint || echo \"warning: Failed to download SwiftLint\"; exit 0\n  ${PROJECT_DIR}/swiftlint \nfi\n";
```
Comment (evnik) [inline FlipperZero/FlipperZero.xcodeproj/project.pbxproj:398]: Could you please move the script into a separate file, so it'll be easier to maintain?

Comment (illabo) [reply FlipperZero/FlipperZero.xcodeproj/project.pbxproj:398]: I was waiting for that. :)
If losing access to project's envvars (eg ${PROJECT_DIR}) isn't an issue I'll eliminate the duplication.

File: FlipperZero/perform_lint.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  swiftlint
+  exit 0
+fi
+# Download a copy of swiftlint if it doesn't exist in project root
+if [ ! -f swiftlint ]; then
+  curl -o- -Ls https://github.com/realm/SwiftLint/releases/latest/download/portable_swiftlint.zip | tar xzf - swiftlint \
+  && chmod +x swiftlint || echo "warning: Failed to download SwiftLint"; exit 0
+fi
+# Lint with local swiftlint
+./swiftlint
```
Comment (evnik) [inline FlipperZero/perform_lint.sh:12]: I'm not sure why, but I don't see any output from the `swiftlint` right after it was downloaded, but it's fine on the next run:
```
$ ./perform_lint.sh
$ ./perform_lint.sh
Linting Swift files at paths 
Linting 'HomeView.swift' (1/4)
Linting 'FlipperZeroApp.swift' (2/4)
Linting 'CoreTests.swift' (3/4)
Linting 'RootView.swift' (4/4)
/.../Flipper-iOS-App/FlipperZero/CoreTests/CoreTests.swift:12:4: warning: Todo Violation: TODOs should be resolved (remove this dummy test when mo...). (todo)
Done linting! Found 1 violation, 0 serious in 4 files.
```
Also, should we add the binary into `.gitignore`?

File: FlipperZero/FlipperZero.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -16,6 +16,7 @@
 		44A5B5B324F05FCE009EE7FB /* Core.framework in Embed Frameworks */ = {isa = PBXBuildFile; fileRef = 44A5B59824F05E74009EE7FB /* Core.framework */; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
 		44A5B5B724F05FD5009EE7FB /* Core.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 44A5B59824F05E74009EE7FB /* Core.framework */; };
 		44A5B5B824F05FD5009EE7FB /* Core.framework in Embed Frameworks */ = {isa = PBXBuildFile; fileRef = 44A5B59824F05E74009EE7FB /* Core.framework */; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
+		497F0FC2252702B7000B2A86 /* perform_lint.sh in Resources */ = {isa = PBXBuildFile; fileRef = 497F0FC1252702B7000B2A86 /* perform_lint.sh */; };
```
[the remark this answers is not in the document]
Comment (illabo) [reply FlipperZero/FlipperZero.xcodeproj/project.pbxproj:19]: My bad. Didn't notice it was auto-added on import to project.

File: FlipperZero/perform_lint.sh (hunk 2 above)
Comment (illabo) [reply FlipperZero/perform_lint.sh:12]: Absolutely it should be ignored. It was already added in #16 
Regarding output, yeah it is also my fault. Gonna fix now.

File: FlipperZero/perform_lint.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Lint with systemwide installation if present
+if which swiftlint >/dev/null; then
```
[the remark this answers is not in the document]
Comment (illabo) [reply FlipperZero/perform_lint.sh:2]: I agree with the reasoning in [SC2230](https://github.com/koalaman/shellcheck/wiki/SC2230) but is it really important? It is available on any macOS setup able to run Xcode 12. ¯\_(ツ)_/¯

**Review by evnik — APPROVED** [review]
There is an [issue](https://github.com/realm/SwiftLint/issues/1392#issuecomment-302565650) with `xcpretty` hiding SwiftLift warnings (not sure about errors), but let's address it separately

**Comment by illabo** [comment]
@evnik There are several ways to deal with `xcpretty` problem. 
We may run `swiftlint` prior `xcodebuild` and it would be great for use in actions with specialized reporter: `swiftlint lint --reporter github-actions-logging`. This way we'll get well formatted report separately from build log. 
To prevent running linter twice I can add envvar check to the build phase script. So we may set `env` in workflow `steps`, e.g. `XC_SKIPLINT: true` and check this var in script. 
Alternatively we may drop `xcpretty` altogether (it's in Ruby, it's unmaintained for two years) and use `xcbeautify` instead ([please check its repo](https://github.com/thii/xcbeautify), fresh faster tool in Swift). 
Or, maybe, we could do both things.

**Pull request merged**